### ISSUE TITLE
TLS Termination

### ISSUE THREAD

**Issue by regbo** [marianobarrios/tls-channel#16]
I currently use HAProxy in front of RabbitMQ and MongoDB to do TLS termination. I then use a java based load balancer to handle http termination for websocket. I prefer the Java route because it is more flexible, and would love to combine the two.

Let's say I have an insecure server running on tcp://host_foo:5555. Is it possible to use this library to perform TLS termination on tls://host_bar:6555 and then insecurely tunnel the traffic to host_foo?

In other words:

[xyc_client] <-- secure tcp (tls) --> [tls-channel-server] <-- insecure tcp --> [xyc_server]

**Comment by regbo** [comment]
Hey I sort of answered my own question here:

https://github.com/regbo/java-nio-tls-proxy

By modifying another project I was able to get the above scenario to work. If you an think of a better way to do it, I think it would be a nice addition to your project.

**Comment by marianobarrios** [comment]
Answering your original question, yes, that is a valid use case. Actually, originally the library was extracted from a HTTP load balancer.
